src/Components/Web.JS/src/Services/NavigationManager.ts

```diff
@@ -60,46 +60,64 @@ export function attachToEventDelegator(eventDelegator: EventDelegator) {
 
       if (isWithinBaseUriSpace(absoluteHref)) {
         event.preventDefault();
-        performInternalNavigation(absoluteHref, true);
+        performInternalNavigation(absoluteHref, /* interceptedLink */ true, /* replace */ false);
       }
     }
   });
 }
 
-export function navigateTo(uri: string, forceLoad: boolean, replace: boolean = false) {
+// For back-compat, we need to accept multiple overloads
+export function navigateTo(uri: string, options: NavigationOptions): void;
+export function navigateTo(uri: string, forceLoad: boolean): void;
+export function navigateTo(uri: string, forceLoad: boolean, replace: boolean): void;
+export function navigateTo(uri: string, forceLoadOrOptions: NavigationOptions | boolean, replaceIfUsingOldOverload: boolean = false) {
```

Comment:
I haven't seen this used in a while in TS 😄.

I believe `navigateTo` ends up in our internal namespace. The only thing that uses that overload is the auth bits in one place. I'm happy if you update those and avoid having to deal with the special case here, since we can consider this API internal and for our consumption, akin to private reflection.

Reply:
It's actually not within internal, however it's [here](https://github.com/dotnet/aspnetcore/blob/main/src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticatorViewCore.cs#L348)

Reply:
`Blazor.navigateTo` (the JS API) is very much public and supported. It's the only way to trigger a client-side Blazor navigation from JS code, so is certainly needed.

Reply:
Yeah, I saw that later. If you want to update the auth package as part of this change to rely on that, It would be nice.